programs/uxd/src/instructions/credix_lp/rebalance_request_create_from_credix_lp_depository.rs

```diff
@@ -0,0 +1,283 @@
+                &ctx.accounts.identity_depository,
+                &ctx.accounts.mercurial_vault_depository,
+                &ctx.accounts.depository,
+            )?;
+        if redeemable_amount_under_management < redeemable_amount_under_management_target_amount {
+            0
+        } else {
+            redeemable_amount_under_management
+                .checked_sub(redeemable_amount_under_management_target_amount)
+                .ok_or(UxdError::MathError)?
+        }
+    };
+
+    let profits_collateral_amount = {
+        let liquidity_collateral_amount_before: u64 =
+            ctx.accounts.credix_liquidity_collateral.amount;
+        let outstanding_collateral_amount_before: u64 = ctx
+            .accounts
+            .credix_global_market_state
+            .pool_outstanding_credit;
+        let total_shares_supply_before: u64 = ctx.accounts.credix_shares_mint.supply;
+        let total_shares_value_before: u64 = liquidity_collateral_amount_before
+            .checked_add(outstanding_collateral_amount_before)
+            .ok_or(UxdError::MathError)?;
+        let owned_shares_amount_before: u64 = ctx.accounts.depository_shares.amount;
+        let owned_shares_value_before: u64 = compute_value_for_shares_amount_floor(
+            owned_shares_amount_before,
+            total_shares_supply_before,
+            total_shares_value_before,
+        )?;
+        owned_shares_value_before
+            .checked_sub(redeemable_amount_under_management)
+            .ok_or(UxdError::MathError)?
+    };
+
+    // ---------------------------------------------------------------------
+    // -- Phase 3
+    // -- We want to withdraw the sum of the profits and collateral at the same time
+    // -- We just create the credix withdraw request with the computed withdrawal amount
+    // ---------------------------------------------------------------------
+
+    let requested_collateral_amount = overflow_value
+        .checked_add(profits_collateral_amount)
+        .ok_or(UxdError::MathError)?;
+    msg!(
+        "[rebalance_request_create_from_credix_lp_depository:requested_collateral_amount:{}]",
+        requested_collateral_amount
+    );
+    credix_client::cpi::create_withdraw_request(
+        ctx.accounts
+            .into_create_withdraw_request_from_credix_lp_context()
+            .with_signer(depository_pda_signer),
+        requested_collateral_amount,
+    )?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 4
+    // -- Emit resulting event
+    // ---------------------------------------------------------------------
+
+    emit!(RebalanceRequestCreateFromCredixLpDepositoryEvent {
+        controller_version: ctx.accounts.controller.load()?.version,
+        depository_version: ctx.accounts.depository.load()?.version,
+        controller: ctx.accounts.controller.key(),
+        depository: ctx.accounts.depository.key(),
+        overflow_value,
```

Comment:
would also emit the `requested_collateral_amount`

Reply:
Sounds good, changing now!